### PR TITLE
[libjpeg-turbo] Enable NEON intrinsics for ARM builds

### PR DESCRIPTION
**Pull request by ZacWalk**
**Describe the pull request**

- #### What does your PR fix?
Enables NEON Intrinsics for ARM builds to improve performance. Performance testing using tjbench:

| config | Snapdragon 3ghz Windows device **Without** Intrinsics | Snapdragon 3ghz Windows device  **With** Intrinsics | Intel i7 11700 3.6ghz Windows device |
| --- | --- | --- | --- |
| x86 | 113 Megapixels/sec (Emulated) | 113 Megapixels/sec (Emulated) | 229 Megapixels/sec | 
| x64 | 121 Megapixels/sec (Emulated) | 121 Megapixels/sec (Emulated) | 278 Megapixels/sec |
| arm64 | 102 Megapixels/sec | **183 Megapixels/sec** | n/a |

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
  No - not needed.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
  Yes

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
  Yes

**If you are still working on the PR, open it as a Draft: https://github.blog/2019-02-14-introducing-draft-pull-requests/**

### CODE REVIEW

File: ports/libjpeg-turbo/portfile.cmake [hunk 1]
```diff
@@ -15,7 +15,7 @@ vcpkg_from_github(
 )
 
 if(VCPKG_TARGET_ARCHITECTURE STREQUAL "arm" OR VCPKG_TARGET_ARCHITECTURE STREQUAL "arm64" OR (VCPKG_CMAKE_SYSTEM_NAME AND NOT VCPKG_CMAKE_SYSTEM_NAME STREQUAL "WindowsStore"))
-    set(LIBJPEGTURBO_SIMD -DWITH_SIMD=OFF)
+    set(LIBJPEGTURBO_SIMD -DWITH_SIMD=ON -DNEON_INTRINSICS=ON)
```
Comment (Neumann-A) [inline ports/libjpeg-turbo/portfile.cmake:18]: The question here is if `NEON_INTRNSICS` are valid on all arm architectures or not.

Comment (ZacWalk) [reply ports/libjpeg-turbo/portfile.cmake:18]: There are runtime checks to determine if SIMD versions of routines should be called (at runtime).
Looking at the intrinsics used, they look compatible with ARMv7 and ARMv8.

Comment (BillyONeal) [reply ports/libjpeg-turbo/portfile.cmake:18]: This is consistent with what other ports do.

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**